From 004153c4384ab57345676f77edbcef7bb4eff0f8 Mon Sep 17 00:00:00 2001
From: Alessandro Astone <ales.astone@gmail.com>
Date: Tue, 6 Oct 2020 14:50:06 +0200
Subject: [PATCH] fixup! Bring back file-based OTA edify functions

(1) install.cpp is now used to build a host library as well,
    and glibc's xattr.h does not include the kernel header
    unlike its bionic counterpart, so manually include it.

(2) UpdaterRuntimeInterface is also implemented by
    SimulatorRuntime for which we would need to define the
    sehandle() function.
    Provide a default implementation which returns null.

Change-Id: If458678b87f871a9b3fd2269d9299ed9eb190c7f
---
 edify/include/edify/updater_runtime_interface.h | 4 +++-
 updater/install.cpp                             | 2 ++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/edify/include/edify/updater_runtime_interface.h b/edify/include/edify/updater_runtime_interface.h
index 66bc1d66..054b85de 100644
--- a/edify/include/edify/updater_runtime_interface.h
+++ b/edify/include/edify/updater_runtime_interface.h
@@ -77,5 +77,7 @@ class UpdaterRuntimeInterface {
   // On devices supports A/B, add current slot suffix to arg. Otherwise, return |arg| as is.
   virtual std::string AddSlotSuffix(const std::string_view arg) const = 0;
 
-  virtual struct selabel_handle* sehandle() const = 0;
+  virtual struct selabel_handle* sehandle() const {
+    return nullptr;
+  }
 };
diff --git a/updater/install.cpp b/updater/install.cpp
index 46f42a52..bd5c966f 100644
--- a/updater/install.cpp
+++ b/updater/install.cpp
@@ -35,6 +35,8 @@
 #include <unistd.h>
 #include <utime.h>
 
+#include <linux/xattr.h>
+
 #include <memory>
 #include <string>
 #include <vector>
-- 
2.15.0

